pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -135,6 +154,28 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 	return observedConfig, errs
 }
 
+// isCloudProviderExternal is used to determine if the cluster should use external cloud providers.
+// Currently, this is opt in via a feature gate. If no feature gate is present, the cluster should remain
+// using the in-tree implementation.
+func (c *cloudProviderObserver) isCloudProviderExternal(listers InfrastructureLister, platform configv1.PlatformType, recorder events.Recorder) (bool, error) {
+	featureGate, err := listers.FeatureGateLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		// No feature gate is set, therefore cannot be external.
+		// This is not an error as the feature gate is an optional resource.
+		recorder.Eventf("ObserveCloudProviderNames", "Optional featuregate.%s/cluster not found", configv1.GroupName)
```

Comment:
deeply nested events in getters are a bad pattern. In other words: don't put side-effects into getters.